subctl/issue-196-disbale-features-in-build.md

```diff
@@ -0,0 +1,95 @@
+# Disable certain features while building `subctl`
+
+Related Issue:
+[Support disabling certain features in builds](https://github.com/submariner-io/enhancements/issues/196)
+
+## Summary
+
+Some builds of `subctl` (e.g. for OCM) don’t need certain sub-commands such as `deploy-broker`, `join` etc. It would be useful to be able
+to disable these features so that users don’t even see them in `--help` output.
+
+## Background
+
+Snippet from [Go's official documentation](https://pkg.go.dev/cmd/go#hdr-Build_constraints) to under Go constraints.
+
+    Go provides build constraints to identify files that should be included in the package.
+    Build constraints are given by a line comment that begins
+    
+        //go:build
+    
+    Constraints may appear in any kind of source file (not just Go), but they must appear near the top of the file, preceded only by blank
+    lines and other line comments. These rules mean that in Go files a build constraint must appear before the package clause.
+    
+    To distinguish build constraints from package documentation, a build constraint should be followed by a blank line.
+    
+    A build constraint comment is evaluated as an expression containing build tags combined by ||, &&, and ! operators and parentheses.
+    Operators have the same meaning as in Go.
+
+## Proposal
+
+All the sub-commands that are related to deployment would be excluded from builds by adding negated tags via build constraints. The
+downstream build of `subctl` would need to be built using the `tags` flag.
+
+## Design
+
+The following `subctl` sub-commands will have build constraint so that they can be excluded from the build based on tags provided at
```

Comment:
```suggestion
The following `subctl` sub-commands will have build constraints so that they can be excluded from the build based on tags provided at
```